nucypher/network/nodes.py

```diff
@@ -133,6 +137,9 @@ def nickname(self):
             self._nickname = Nickname.from_seed(self.checksum_address)
         return self._nickname
 
+    def rest_url(self):
+        return self.rest_interface.uri
```

Comment:
This was already being stored, but just not previously made accessible?

Reply:
Yes, before this method was only present for an Ursula, so it required a `mature()` call at some point. So in #2574 the call that was now being removed was added. But as it seems, the required data is already present in a sprout, so we can just expose it.

That said, I still think there we need to review this approach and make the maturation event less random, to provide some guarantees for the users of `FleetSensor`. This will make testing easier, too.

Reply:
> That said, I still think there we need to review this approach and make the maturation event less random, to provide some guarantees for the users of FleetSensor. This will make testing easier, too.

👍 - perhaps we can explore as a follow-up issue.

Reply:
Filed #2715 - it's more specific, but changes to how `mature()` is called will probably be a part of the solution.